internal/http/routes.go

```diff
@@ -15,13 +14,15 @@ func useRoutes(handler handler.Handler) *chi.Mux {
 		middleware.Logger,
 	)
 
-	r.Post("/", handler.Post)
+	r.With(middleware.AuthMiddleware()).Post("/", handler.Post)
 	r.Get("/{id}", handler.Get)
 	r.Get("/ping", handler.Ping)
 
 	r.Route("/api", func(r chi.Router) {
-		r.Post("/shorten", handler.PostAPI)
-		r.Post("/shorten/batch", handler.PostBatch)
+		r.With(middleware.AuthMiddleware()).Post("/shorten", handler.PostAPI)
+		r.With(middleware.AuthMiddleware()).Post("/shorten/batch", handler.PostBatch)
+		r.With(middleware.AuthCheckMiddleware()).Get("/user/urls", handler.GetAll)
+		r.With(middleware.AuthCheckMiddleware()).Delete("/user/urls", handler.Delete)
```

Comment:
Можно группировать дальше:
```suggestion
                r.Route("/shorten", func(r chi.Router) {
                    r.With(middleware.AuthMiddleware()).Post("/", handler.PostAPI)
                    r.Post("/batch", handler.PostBatch)
                })
                r.Route("/user", func(r chi.Router) {
                    r.With(middleware.AuthCheckMiddleware()).Get("/urls", handler.GetAll)
                    r.Delete("/urls", handler.Delete)
                })
```